controllers/argocd/applicationset.go

```diff
@@ -455,3 +461,52 @@ func setAppSetLabels(obj *metav1.ObjectMeta) {
 	obj.Labels["app.kubernetes.io/part-of"] = "argocd-applicationset"
 	obj.Labels["app.kubernetes.io/component"] = "controller"
 }
+
+// reconcileApplicationSetService will ensure that the Service is present for the ApplicationSet webhook and metrics component.
+func (r *ReconcileArgoCD) reconcileApplicationSetService(cr *argoprojv1a1.ArgoCD) error {
+	log.Info("reconciling applicationset service")
+
+	svc := newServiceWithSuffix(common.ApplicationSetServiceNameSuffix, common.ApplicationSetServiceNameSuffix, cr)
+	if cr.Spec.ApplicationSet == nil {
+		// if ensureAutoTLSAnnotation(svc, common.ArgoCDServerTLSSecretName, cr.Spec.Server.WantsAutoTLS()) {
+		// 	return r.Client.Update(context.TODO(), svc)
+		// }
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, svc) {
+			err := argoutil.FetchObject(r.Client, cr.Namespace, svc.Name, svc)
+			if err != nil {
+				return err
+			}
+			log.Info(fmt.Sprintf("Deleting applicationset controller service %s as applicationset is disabled", svc.Name))
+			err = r.Delete(context.TODO(), svc)
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		if argoutil.IsObjectFound(r.Client, cr.Namespace, svc.Name, svc) {
+			return nil // Service found, do nothing
+		}
```

Comment:
I'm not sure whether that should be the correct reconciliation behavior. I know it's probably similar to other services we reconcile, but we really should think about it. The behavior here will only ever create the service, but if we want to make changes to the service in the code, it will never adapt unless the service is actually deleted.